core/common-util/src/main/java/uk/gov/gchq/gaffer/commonutil/iterable/LimitedCloseableIterator.java

```diff
@@ -52,12 +64,18 @@ public void close() {
 
     @Override
     public boolean hasNext() {
-        boolean hasNext = (null == end || index < end) && iterator.hasNext();
-        if (!hasNext) {
+        boolean hasNext = (null == end || index < end);
```

Comment:
Please could you rename this hasNext variable to withinLimit or something similar.